src/LatencyEstimator.hpp

```diff
@@ -20,11 +22,13 @@ namespace signal_estimator {
 // detects strikes in output and input signals and calculates the latency between them
 class LatencyEstimator : public IEstimator {
 public:
-    LatencyEstimator(const Config& config);
+    LatencyEstimator(const Config& config, std::unique_ptr<IFormatter>& formatter);
```

Comment:
Here and in LossEstimator.

Passing unique_ptr by non-constant reference assumes that the callee (ctor) is allowed modify the passed pointer, e.g. reset it to another value.

Since it's not the case, I suggest to replace it with either of these:

* passing unique_ptr by value
* passing unique_ptr by constant reference
* passing raw reference IFormatter